internal/provider/resource_mackerel_dashboard.go

```diff
@@ -0,0 +1,576 @@
+				Description: schemaDashboardGraph_rangeRelativeDesc,
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.ExactlyOneOf(
+						path.MatchRelative(),
+						path.MatchRelative().AtParent().AtName("absolute"),
+					),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"period": schema.Int64Attribute{
+							Description: schemaDashboardGraph_rangeRelativePeriodDesc,
+							Required:    true,
+						},
+						"offset": schema.Int64Attribute{
+							Description: schemaDashboardGraph_rangeRelativeOffsetDesc,
+							Required:    true,
+						},
+					},
+				},
+			},
+			"absolute": schema.ListNestedBlock{
+				Description: schemaDashboardGraph_rangeAbsoluteDesc,
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"start": schema.Int64Attribute{
+							Description: schemaDashboardGraph_rangeAbsoluteStartDesc,
+							Required:    true,
+						},
+						"end": schema.Int64Attribute{
+							Description: schemaDashboardGraph_rangeAbsoluteEndDesc,
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+const (
+	schemaDashboardGraphDesc      = "The graph widget."
+	schemaDashboardGraph_nameDesc = "The name of the graph."
+
+	schemaDashboardGraph_hostDesc        = "The host graph."
+	schemaDashboardGraph_host_hostIDDesc = "The ID of the host."
+
+	schemaDashboardGraph_roleDesc              = "The role graph."
+	schemaDashboardGraph_role_roleFullNameDesc = "The service name or role ID."
+	schemaDashboardGraph_role_isStackedDesc    = "Whether the graph is stacked or line graph."
+
+	schemaDashboardGraph_serviceDesc             = "The service graph."
+	schemaDashboardGraph_service_serviceNameDesc = "The name of the service."
+
+	schemaDashboardGraph_expressionDesc            = "The expression graph."
+	schemaDashboardGraph_expression_expressionDesc = "The expression representing the graph."
+
+	schemaDashboardGraph_queryDesc        = "The query graph."
+	schemaDashboardGraph_query_queryDesc  = "The PromQL-style query."
+	schemaDashboardGraph_query_legendDesc = "The query legend."
+)
+
+var schemaDashboardResource_graph = schema.ListNestedBlock{
+	NestedObject: schema.NestedBlockObject{
+		Attributes: map[string]schema.Attribute{
+			"title": schemaDashboardResource_widgetTitle,
+		},
+		Blocks: map[string]schema.Block{
+			"range":  schemaDashboardResource_widgetRange,
+			"layout": schemaDashboardResource_widgetLayout,
+
+			"host": schema.ListNestedBlock{
+				Description: schemaDashboardGraph_hostDesc,
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.ExactlyOneOf(
+						path.MatchRelative(),
+						path.MatchRelative().AtParent().AtName("role"),
+						path.MatchRelative().AtParent().AtName("service"),
+						path.MatchRelative().AtParent().AtName("service"),
```

Comment:
Is this an intentional duplication?

Reply:
Not intentional. Fixed it: https://github.com/mackerelio-labs/terraform-provider-mackerel/pull/239/commits/15d4a2014c7f568ded7376ed8eff59af54c006b8